android/app/src/main/java/BitcoinJ/EncryptorAES.java

```diff
@@ -0,0 +1,48 @@
+package BitcoinJ;
+
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.InvalidParameterSpecException;
+import java.util.Arrays;
+import java.util.Base64;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+public class EncryptorAES {
+
+    public static byte[] encrypt(byte[] key, byte[] clear) throws Exception
+    {
+        MessageDigest md = MessageDigest.getInstance("md5");
+        byte[] digestOfPassword = md.digest(key);
```

Comment:
It's mandatory to use `md5`? MD5 is broken and might be an security issue. If we can use sha2, sha3 or scrypt it would be significantly better

Reply:
it's a key digest, `md5` is not mandatory , a single round of sha2 is perfect .
According to:
- https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html
`MessageDigest md = MessageDigest.getInstance("md5");`  can be changed by: `MessageDigest md = MessageDigest.getInstance("SHA-256");`